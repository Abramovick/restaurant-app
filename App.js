import React, { Component } from 'react';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';
import { AppLoading, Asset, Font } from 'expo';
import { Ionicons } from '@expo/vector-icons';
import { combineReducers, createStore } from 'redux';
import { ApolloProvider } from 'react-apollo';
import ApolloClient from 'apollo-client';
import { HttpLink } from 'apollo-link-http';
import { reducer as formReducer } from 'redux-form';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SubscriptionClient } from 'subscriptions-transport-ws';
import { addGraphQLSubscriptions } from 'add-graphql-subscriptions';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import WebSocket from 'ws';
import RootNavigation from './app/navigation/RootNavigation';
import robotDev from './app/assets/images/robot-dev.png';
import robotProd from './app/assets/images/robot-prod.png';
import spaceMono from './app/assets/fonts/SpaceMono-Regular.ttf';
import allerLight from './app/assets/fonts/aller-light.ttf';
import OpenSansLight from './app/assets/fonts/OpenSans-Light.ttf';
import SinkinSans100Thin from './app/assets/fonts/SinkinSans-100Thin.ttf';
import SinkinSans200XLight from './app/assets/fonts/SinkinSans-200XLight.ttf';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  statusBarUnderlay: {
    height: 24,
    backgroundColor: 'rgba(0,0,0,0.2)',
  },
});

const wsClient = new SubscriptionClient('wss://subscriptions.graph.cool/v1/cja8r6dcn33ui0154xhdkhlel', {
  reconnect: true,
  connectionParams: {
  }
}, WebSocket);
const uri = 'https://api.graph.cool/simple/v1/cja8r6dcn33ui0154xhdkhlel';

const httpLink = new HttpLink({ uri });

const linkWithSubscriptions = addGraphQLSubscriptions(
  httpLink,
  wsClient
);

const client = new ApolloClient({
  link: linkWithSubscriptions,
  cache: new InMemoryCache().restore(window.__APOLLO_STATE__),
});

const store = createStore(combineReducers({
  form: formReducer
}));
export default class App extends Component {
  static propTypes = {
    skipLoadingScreen: PropTypes.bool,
  }
  state = {
    isLoadingComplete: false,
  };
  _loadResourcesAsync = async () => Promise.all([
    Asset.loadAsync([
      robotDev,
      robotProd,
    ]),
    Font.loadAsync({
      ...Ionicons.font,
      'space-mono': spaceMono,
      'aller-light': allerLight,
      'OpenSans-Light': OpenSansLight,
      'SinkinSans-100Thin': SinkinSans100Thin,
      'SinkinSans-200XLight': SinkinSans200XLight,
    }),
  ]);

  _handleLoadingError = (error) => {
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    }
    return (
      <Provider store={store}>
        <ApolloProvider client={client}>
          <View style={styles.container}>
            {Platform.OS === 'ios' && <StatusBar barStyle="default" />}
            {Platform.OS === 'android' && <View style={styles.statusBarUnderlay} />}
            <RootNavigation />
          </View>
        </ApolloProvider>
      </Provider>
    );
  }
}
