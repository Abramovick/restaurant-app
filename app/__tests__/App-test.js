import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../App';

jest.mock('../navigation/RootNavigation', () => 'RootNavigation');


it('renders the loading screen', async () => {
  const tree = renderer.create(<App />);
  expect(tree.getInstance()._loadResourcesAsync()).toMatchSnapshot();
  expect(tree.getInstance()._handleLoadingError()).toMatchSnapshot();
  expect(tree.getInstance()._handleFinishLoading()).toMatchSnapshot();

  expect(tree).toMatchSnapshot();
});

it('renders the root without loading screen', async () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the status bar if the platform is ios', async () => {
  jest.resetModules();
  jest.doMock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'ios';
    return Platform;
  });

  const tree = renderer.create(<App skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the view if the platform if android', async () => {
  jest.resetModules();
  jest.doMock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'android';
    return Platform;
  });
  const tree = renderer.create(<App skipLoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
