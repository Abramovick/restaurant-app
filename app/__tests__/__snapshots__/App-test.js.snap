// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the loading screen 1`] = `
Promise {
  "_40": 0,
  "_55": null,
  "_65": 0,
  "_72": null,
}
`;

exports[`renders the loading screen 2`] = `undefined`;

exports[`renders the loading screen 3`] = `undefined`;

exports[`renders the loading screen 4`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RootNavigation />
</View>
`;

exports[`renders the root without loading screen 1`] = `<ExponentAppLoading />`;

exports[`renders the status bar if the platform is ios 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RootNavigation />
</View>
`;

exports[`renders the view if the platform if android 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.2)",
        "height": 24,
      }
    }
  />
  <RootNavigation />
</View>
`;
